services/debugger/uploader.go

```diff
@@ -57,25 +61,35 @@ func New(url string, transformer Transformer) UploaderI {
 	eventBuffer := make([]interface{}, 0)
 	client := &http.Client{}
 
-	uploader := &Uploader{url: url, transformer: transformer, eventBatchChannel: eventBatchChannel, eventBuffer: eventBuffer, Client: client}
+	uploader := &Uploader{url: url, transformer: transformer, eventBatchChannel: eventBatchChannel, eventBuffer: eventBuffer, Client: client, bgWaitGroup: sync.WaitGroup{}}
 	uploader.Setup()
 	return uploader
 }
 
 func (uploader *Uploader) Start() {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Can we potentially have `Start()` accept a `context.Context` as a parameter? I think it could play a very important role in the scope of this PR. WDYT?

Reply:
So that we can use the ctx to match the Start() with the Stop()?

Reply:
We wouldn't even need `Stop()`, right? Plus it should make the whole thing easier to test.

Reply:
The chain of events as it is now is: 
Stop() -> stop(channel) -> handleEvents() loop exit -> ctx cancel -> flushEvents trigger -> flush events (and return).
So, we need stop to stop the channel and trigger this.
Unless you mean that we can remove stop() altogether and use the cancel() in the test scope?

Reply:
I meant that we can just pass `ctx` to `Start()` and use that to stop everything. If the context gets canceled the uploader stops everything. No need to `Stop()`, correct?

Reply:
As discussed we're going to follow this guideline https://www.rudderstack.com/blog/implementing-graceful-shutdown-in-go/